### PR TITLE
feat(ci): auto-assign github PR reviewers and assignee using github-action

### PR DESCRIPTION
**Pull request by prateekpandey14**
Use a GitHub Action to automatically assign GitHub PRs to the author, as
well as add reviewers.

Signed-off-by: prateekpandey14 <prateek.pandey@mayadata.io>

### CODE REVIEW

File: .github/workflows/auto_assign_prs.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: 'Auto Assign PRs'
```
Comment (kmova) [inline .github/workflows/auto_assign_prs.yml:1]: this is probably not required. just go with .github/auto_assign.yaml

File: .github/auto_assign.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+addReviewers: true
+
+addAssignees: author
+
+# Set this to the length of the reviewers list
+# because the default was not including everyone
+numberOfReviewers: 3
```
Comment (kmova) [inline .github/auto_assign.yaml:7]: make this 2.

File: .github/auto_assign.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+addReviewers: true
+
+addAssignees: author
```
Comment (kmova) [inline .github/auto_assign.yaml:3]: keep this as false.

File: .github/workflows/auto_assign_prs.yml (hunk 1 above)
Comment (prateekpandey14) [reply .github/workflows/auto_assign_prs.yml:1]: as per the docs here https://github.com/marketplace/actions/auto-assign-action they suggest to have 2 files

Comment (prateekpandey14) [reply .github/workflows/auto_assign_prs.yml:1]: let me try .. if it works since the name is same .

File: .github/auto_assign.yaml (hunk 3 above)
Comment (prateekpandey14) [reply .github/auto_assign.yaml]: done

File: .github/workflows/auto_assign_prs.yml (hunk 1 above)
Comment (prateekpandey14) [reply .github/workflows/auto_assign_prs.yml:1]: No, it needs separate configuration file for the auto-assign action

Comment (kmova) [reply .github/workflows/auto_assign_prs.yml:1]: It is working without the workflow here: https://github.com/openebs/Mayastor/tree/develop/.github

Comment (prateekpandey14) [reply .github/workflows/auto_assign_prs.yml:1]: right , i was looking for more example like this to better mange the PRs based on the types
 here https://github.com/vmware-tanzu/velero/blob/main/.github/workflows/auto_assign_prs.yml .
 
Should we go with this way or ?

Comment (kmova) [reply .github/workflows/auto_assign_prs.yml:1]: I see .. and this gives the flexibility to add auto-labeling job as well.

**Review by kmova — APPROVED** [review]

**Pull request merged**